src/fu-engine.c

```diff
@@ -1107,11 +1170,23 @@ fu_engine_add_device_flag(FuEngine *self,
 				    fu_device_get_id(proxy));
 			return FALSE;
 		}
+		if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_INTERNAL)) {
+			fu_device_add_flag(device, flag);
+			if (!fu_engine_add_device_flag_emulation_tag_internal(
+				self,
+				fu_device_get_id(device),
+				error))
+				return FALSE;
+			fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+			return TRUE;
+		}
 		g_hash_table_insert(self->emulation_ids,
 				    g_strdup(fu_device_get_id(device)),
 				    GUINT_TO_POINTER(1));
 		fu_engine_ensure_context_flag_save_events(self);
-		fu_engine_emit_device_request_replug_and_install(self, device);
+		if (!fu_device_has_flag(device, FWUPD_DEVICE_FLAG_INTERNAL))
```

Comment:
I still don't see how we can get here -- if we're FWUPD_DEVICE_FLAG_INTERNAL then we got handled in the block above.

Reply:
Oh!  It's because I tagged an internal device that I expected had INTERNAL, but it's missing it.  Totally a different problem.  Will revert this one.